server/plugin/oAuth.go

```diff
@@ -173,6 +219,22 @@ func (p *Plugin) GenerateOAuthToken(code, state string) error {
 	return nil
 }
 
+// isAccessTokenExpired checks if a user's access token is expired
+func (p *Plugin) isAccessTokenExpired(mattermostUserID string) bool {
+	user, err := p.Store.LoadUser(mattermostUserID)
+	if err != nil {
+		p.API.LogError(constants.ErrorLoadingUserData, "Error", err.Error())
+		return false
+	}
+
+	// TODO: use middleware for all such places to check if user's oAuth is completed
+	if user.AccessToken != "" && user.ExpiresAt.Before(time.Now()) {
```

Comment:
have a buffer of some time like check whether the token is expiring in 1 min if it is refresh that